cytotable/sources.py

```diff
@@ -134,21 +144,36 @@ def _get_source_filepaths(
 
     # group files together by similar filename for later data operations
     grouped_sources = {}
-    for unique_source in set(source["source_path"].name for source in sources):
-        grouped_sources[unique_source.capitalize()] = [
-            # case for files besides sqlite
-            source if source["source_path"].suffix.lower() != ".sqlite"
-            # if we have sqlite entries, update the source_path to the parent
-            # (the parent table database file) as grouped key name will now
-            # encapsulate the table name details.
-            else {
-                "source_path": source["source_path"].parent,
-                "table_name": source["table_name"],
-            }
-            for source in sources
-            # focus only on entries which include the unique_source name
-            if source["source_path"].name == unique_source
-        ]
+
+    # if we have no targets, create a single group inferred from a common prefix and suffix
```

Comment:
can you provide a bit more detail about scenarios where we would expect no targets?

Reply:
Thank you, will add this in here. Generally I felt this might be involved if there was uncertainty or a lack of specification about compartments or metadata provided from a data source. In terms of the user, this might be a "simplest case" of using CytoTable with a directory that contains only one compartment or metadata related files. This might be a bad assumption on my part. Perhaps this will relate to the compartment + metadata validation step in #136 .

Reply:
Added more to the comments here within the code to better describe this.